code/frameworks/angular/src/client/docs/compodoc.ts

```diff
@@ -256,7 +263,7 @@ export const extractArgTypesFromData = (componentData: Class | Directive | Injec
           category: section,
           type: {
             summary: isMethod(item) ? displaySignature(item) : item.type,
-            required: isMethod(item) ? false : !item.optional,
+            required: isMethod(item) ? false : item.required,
```

Comment:
I was going to ask if this should be `required: isMethod(item) ? false : isRequired(item),` to avoid `undefined`, but I don't know if anything even uses this. I thought it used to do something, but I can't find where and the other frameworks arg type extract functions don't seem to set it. Maybe if I dug some more I could find why it is here, but if the other frameworks aren't setting it then maybe it doesn't even need to be here.

Reply:
You're right, i'm not sure it is even used. If i set that prop manually in a story, i.e:

```
argTypes: {
  title: {
     table: {
          type: { required: true },
      },
  }
}
```

It does absolutely nothing.

However manually setting it this way:

```
argTypes: {
  title: {
     type: { name: 'string', required: true },
  }
}
```

renders the correct asterix in red to say it's required (which was the intention of the PR)

That gets fixed by this change in the PR:

`return { name: compodocType, ...setRequiredProperty(property) };` (line 153)

So i'm happy to either refactor to as you mentioned above, or remove entirely as part of the PR if we're confident it's no longer used - happy to take your steer on that :)

Reply:
@Marklb You're right, i'm not sure it is even used. If i set that prop manually in a story, i.e:

```
argTypes: {
  title: {
     table: {
          type: { required: true },
      },
  }
}
```

It does absolutely nothing.

However manually setting it this way:

```
argTypes: {
  title: {
     type: { name: 'string', required: true },
  }
}
```

renders the correct asterix in red to say it's required (which was the intention of the PR)

That gets fixed by this change in the PR:

`return { name: compodocType, ...setRequiredProperty(property) };` (line 153)

So i'm happy to either refactor to as you mentioned above, or remove entirely as part of the PR if we're confident it's no longer used - happy to take your steer on that :)